tests/extension/test_app_extension.py

```diff
@@ -378,5 +378,28 @@ def index():
             self.assertEqual(resp.status_code, 200)
 
 
+class AppExtensionPreflight(FlaskCorsTestCase):
+    def test_preflight(self):
+        ''' Ensure that view function methods override app level defaults '''
+        self.app = Flask(__name__)
+        CORS(self.app)
+
+        @self.app.route('/')
+        def index():
+            return 'Welcome'
+
+        @self.app.route('/test', methods=['POST'])
+        def index2():
+            return 'Welcome 2'
+
+        res = self.preflight('/', 'GET', origin='www.example.com')
```

Comment:
This change is also backward incompatible in a way that instead of default or level configuration, methods from view function are used.